mlrun/runtimes/base.py

```diff
@@ -282,26 +281,26 @@ def _fill_credentials(self):
 
     def run(
         self,
-        runspec: RunObject = None,
-        handler=None,
-        name: str = "",
-        project: str = "",
-        params: dict = None,
-        inputs: Dict[str, str] = None,
-        out_path: str = "",
-        workdir: str = "",
-        artifact_path: str = "",
-        watch: bool = True,
-        schedule: Union[str, mlrun.common.schemas.ScheduleCronTrigger] = None,
-        hyperparams: Dict[str, list] = None,
-        hyper_param_options: HyperParamOptions = None,
-        verbose=None,
-        scrape_metrics: bool = None,
-        local=False,
-        local_code_path=None,
-        auto_build=None,
-        param_file_secrets: Dict[str, str] = None,
-        notifications: List[mlrun.model.Notification] = None,
+        runspec: Optional[Union["mlrun.run.RunTemplate", "mlrun.run.RunObject"]] = None,
```

Comment:
how come we can pass a concrete and one time a template? can you perhaps elaborate on function docstring when such cases occurs?

Reply:
RunTemplate is essentially a RunObject without a status. We generate a RunObject from whatever is given.
We generate a run object from the template if a RunTemplate is given.
It helped me find some other problems - we also allow dict and there was `_create_run_object` in BaseRuntime that we forgot to delete